client/aws/errors.go

```diff
@@ -38,3 +39,8 @@ func IsEmptyAmazonAccountID(err error) bool {
 func IsIAMRoleDuplicateError(err error) bool {
 	return strings.Contains(err.Error(), RoleDuplicate)
 }
+
+// IsRouteDuplicateError checks for duplicate Route errors.
+func IsRouteDuplicateError(err error) bool {
+	return strings.Contains(err.Error(), RouteDuplicate)
+}
```

Comment:
Error matcher is needed for checking for existing routes for the NAT gateway route.